MetaMask/metamask-extension
# feat: preferences controller to base controller v2

## **Description**
In this PR, we want to bring PreferencesController up to date with our latest controller patterns by upgrading to BaseControllerV2.


[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/27398?quickstart=1)

## **Related issues**

Fixes: #25917 

## **Manual testing steps**
Use case 1
1. Install the previous release
2. Complete user onboarding
3. Go to settings and change couple of user settings. For example language, currency and theme.
4. Close and disable MM in the extension
5. Checkout the version with these changes
6. Build and login
7. Make sure, the user preferences set earlier are still there

Use case 2
1. Disable all the MM extensions
2. Install the version with these changes
3. When you click on MM, the default language should be English
4. Complete user onboarding
5. Go to settings and change couple of user settings. For example language, currency and theme.
6. Close and disable and enable the MM in extension. which forces user to login MM in the extension
7. Once you login again, make sure, the user preferences set earlier are still there

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
